source/s3_client.c

```diff
@@ -1168,20 +1168,28 @@ static struct aws_s3_meta_request *s_s3_client_meta_request_factory_default(
     /* Call the appropriate meta-request new function. */
     switch (options->type) {
         case AWS_S3_META_REQUEST_TYPE_GET_OBJECT: {
-            /* If the initial request already has partNumber, the request is not
-             * splittable(?). Treat it as a Default request.
-             * TODO: Still need tests to verify that the request of a part is
-             * splittable or not */
-            if (aws_http_headers_has(initial_message_headers, aws_byte_cursor_from_c_str("partNumber"))) {
-                return aws_s3_meta_request_default_new(
-                    client->allocator,
-                    client,
-                    AWS_S3_REQUEST_TYPE_GET_OBJECT,
-                    content_length,
-                    false /*should_compute_content_md5*/,
-                    options);
+            struct aws_byte_cursor path_and_query;
+
+            if (aws_http_message_get_request_path(options->message, &path_and_query) == AWS_OP_SUCCESS) {
+                /* If the initial request already has partNumber, the request is not
+                 * splittable(?). Treat it as a Default request.
+                 * TODO: Still need tests to verify that the request of a part is
+                 * splittable or not */
+                struct aws_byte_cursor found_string;
+                struct aws_byte_cursor part_number_query_str = aws_byte_cursor_from_c_str("partNumber");
+                if (aws_byte_cursor_find_exact(&path_and_query, &part_number_query_str, &found_string) ==
```

Comment:
find isn't going to cut it. This would match: "rampartNumbering.s3.us-west-2.amazonaws.com"

I'm sure we have query param helpers somewhere in our codebase

Reply:
I tried to find one, but there is only a uri parsing with query param, which requires the whole uri, I can add something to the uri.h to support it

Reply:
https://github.com/awslabs/aws-c-common/pull/1080